### ISSUE TITLE
Programmatically create tasks based on the product of the task executed in the previous pipeline step

### ISSUE THREAD

**Issue by theotheo** [ploomber/ploomber#726]
I would like to understand how to programmatically create tasks based on the product of the task executed in the previous pipeline step. 

For example, `get_data` creates csv-file and I want to create tasks for each row of csv: `process_row_1`, `process_row_2`, .... Accordingly, I have code using the python api that reads a csv-file -- how do I indicate that this csv-file is the product of another task? 
So how do I use the upstream idiom (`upstream['get_data']`)?

I formulate the question a brief, assuming that my case is quite typical. If this assumption of mine is incorrect, then I am ready to supplement the issue with code that more clearly illustrates my request.

**Comment by theotheo** [comment]
Ok, I partially understood. I just can use something like `dag['get_data'].product`. Nice!

But another problem: in this case, my factory-function and generated tasks don't become part of the dag, although it directly depends on the product of the previous task. For example, plot displays them without connecting arrows. Is it possible to do this dependency more explicit?

**Comment by edublancas** [comment]
Hey @theotheo, thanks for your question!

You have a couple of options to do this. Are you using the `pipeline.yaml` or the Python API. We currently do not support this kind of dynamism (create one task per row), since that involves determining the DAG structure at runtime. However, you can pre-define a number of splits (for example: split my data into 10 chunks), and then process them in parallel, which sounds like it is what you're looking for. If using the pipeline.yaml, you can use `grid` for this (here's a complete [example](https://github.com/ploomber/projects/tree/master/cookbook/grid)).

```yaml
  - source: scripts/process.py
    product:
      nb: products/report.html
      model: products/processed.pickle
    grid:
      split: [1, 2, 3, 4, 5]
```

Then, inside your `process.py` you can define which slice to process based on the value of the `split` parameter.

(Note: consider joining our [community](https://ploomber.io/community/), it's a better channel for quick back and forth questions)

**Comment by edublancas** [comment]
closing this due to inactivity, feel free to re-open if you have questions

**Comment by QAston** [comment]
Any plans for supporting this type of task in the future?

**Comment by edublancas** [comment]
we have no immediate plans, but we're open to contributions!
